app/src/main/java/org/openhdp/hdt/ui/tracking/TrackingViewModel.kt

```diff
@@ -133,15 +138,30 @@ class TrackingViewModel @ViewModelInject constructor(
                 state
             }
         }
-        countdown()
     }
 
     fun onDragStarted() {
         job?.cancel()
     }
 
     fun onDragEnded() {
-        // TODO: advance the timers by the time when ticker was off due to dragging
-        countdown()
+
+    }
+
+    fun onCounterAdded(item: AddStopwatchViewState) {
+        val selectedCategoryId = item.categories.firstOrNull { it.selected }?.category?.id ?: return
+
+        viewModelScope.launch(Dispatchers.Main) {
+            runCatching {
+                val count = stopwatchRepository.stopwatchDAO.getAllStopwatchesCount()
+                val stopwatch = Stopwatch(count + 1, item.name, selectedCategoryId)
+                stopwatchRepository.stopwatchDAO.createStopwatch(stopwatch)
```

Comment:
OK, I should work on Repository because here this call should be sufficient: 
`stopwatchRepository.createStopwatch(stopwatch)`